cookbooks/bach_spark/templates/default/spark-defaults.conf.erb

```diff
@@ -0,0 +1,14 @@
+spark.driver.extraLibraryPath=/usr/hdp/current/hadoop-client/lib/native:/usr/hdp/current/hadoop-client/lib/native/Linux-amd64-64
+spark.executor.extraLibraryPath=/usr/hdp/current/hadoop-client/lib/native:/usr/hdp/current/hadoop-client/lib/native/Linux-amd64-64
+spark.executor.extraJavaOptions -verbose:gc -XX:+PrintHeapAtGC -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCDateStamps -XX:+PrintTenuringDistribution -XX:+UseNUMA -XX:+PrintGCApplicationStoppedTime -XX:+UseCompressedOops -XX:+PrintClassHistogram -XX:+PrintGCApplicationConcurrentTime
+spark.shuffle.consolidateFiles true
+spark.shuffle.io.numConnectionsPerPeer 2
+spark.history.fs.logDirectory <%= node['spark']['hdfs_url'] %>/spark-history
+spark.eventLog.dir <%= node['spark']['hdfs_url'] %>/spark-history
+spark.eventLog.enabled true
+spark.logConf true
+spark.dynamicAllocation.enabled true
```

Comment:
One concern here is that if we do not set "spark.dynamicAllocation.maxExecutors" with "spark.dynamicAllocation.enabled", whether or not some big application will eat up the cluster resource.  In my example, my application by default is executed in several shffleMapStages, each stage is made of different number of tasks by default (I think use as much resource as it can). And each task can be assigned to different executors. If one stage has a lot of tasks that invoke the same huge amount of executors in the cluster, this might uses up the resource of cluster, and make other applications fall into dead lock.  Is this concern valid?